tools/tpm2_createprimary.c

```diff
@@ -293,24 +294,26 @@ bool tpm2_tool_onstart(tpm2_options **opts) {
     return *opts != NULL;
 }
 
-int tpm2_tool_onrun(TSS2_SYS_CONTEXT *sapi_context, tpm2_option_flags flags) {
+#define VALID_CTX(ctx) \
+	(ctx.flags.A && ctx.flags.g && ctx.flags.G)
 
+int tpm2_tool_onrun(TSS2_SYS_CONTEXT *sapi_context, tpm2_option_flags flags) {
     UNUSED(flags);
-    int returnVal = 0;
+    int returnVal;
 
-    if(ctx.flags.A == 1 && ctx.flags.g == 1 && ctx.flags.G == 1) {
-        returnVal = create_primary(sapi_context);
+    if (VALID_CTX(ctx) != true) {
```

Comment:
I don't know if this is in the coding standard, but prefer !VALID_CTX(ctx) here.